### PR TITLE
[DS-433] TagGroup api update

### PR DESCRIPTION
**Pull request by vicky-comeau**

### CODE REVIEW

**Review by fraincs — DISMISSED** [review]
Haven't seen anything worth mentioning

File: packages/components/src/tag/src/TagGroup.tsx [hunk 1]
```diff
@@ -128,7 +160,9 @@ function TagGroup(props: TagGroupProps, ref: ForwardedRef<HTMLDivElement>) {
  *
  * [View Documentation](TODO)
  */
-const _TagGroup = forwardRef<HTMLDivElement, TagGroupProps>(TagGroup);
-_TagGroup.displayName = "TagGroup";
+const _TagGroup = forwardRef(TagGroup) as <T>(
```
Comment (alexasselin008) [inline packages/components/src/tag/src/TagGroup.tsx:163]: what is happening here?

File: packages/components/src/tag/tests/jest/TagGroup.test.tsx [hunk 2]
```diff
@@ -1,33 +1,37 @@
 import { render, screen } from "@hopper-ui/test-utils";
 import { createRef } from "react";
 
-import { Tag, TagGroupContext, TagList, TagGroup } from "../../src/index.ts";
+import { Tag, TagGroup, TagGroupContext, type TagListProps } from "../../src/index.ts";
+
+type ExtendedTagListProps<T extends object> = TagListProps<T> & {
```
Comment (alexasselin008) [inline packages/components/src/tag/tests/jest/TagGroup.test.tsx:6]: Why is this needed?

Comment (alexasselin008) [reply packages/components/src/tag/tests/jest/TagGroup.test.tsx:6]: Btw this kind of workaround i'd like you to point them to me before I start reviewing.

File: packages/components/src/tag/src/TagGroup.tsx (hunk 1 above)
Comment (vicky-comeau) [reply packages/components/src/tag/src/TagGroup.tsx:163]: we do this on all the collections or components that have <T> although maybe we can adopt the way spectrum does it (didn't have this before).

File: packages/components/src/tag/tests/jest/TagGroup.test.tsx (hunk 2 above)
Comment (vicky-comeau) [reply packages/components/src/tag/tests/jest/TagGroup.test.tsx:6]: well since the data-testid prop isn't directly on TagList, but is inside tagListProps prop, typescript thinks that it doesn't support data attributes.. so it's basically to shut typescript up without putting a comment to ignore it.
